### ISSUE TITLE
New rule: prefer-true-attribute-shorthand

### ISSUE THREAD

**Issue by g-plane** [vuejs/eslint-plugin-vue#1781]
**Please describe what the rule should do:**

In template, if there's an attribute whose value is `true`, such as `:show="true"`, this rule will warn and provide auto fix (becomes `show`, see the example below). This can reduce verbosity.

**What category should the rule belong to?**

[x] Suggests an alternate way of doing something (suggestion)

**Provide 2-3 code examples that this rule should warn about:**

Invalid:

```vue
<template>
  <my-dialog :show="true" />
</template>
```

which can be fixed to:

```vue
<template>
  <my-dialog show />
</template>
```

**Comment by ota-meshi** [comment]
Thank you for rule proposal.
I would like to use that rule, but by the definition of props they are not equivalent.

```vue
<script>
export default {
  name: 'MyComponent',
  props: {
    a: Boolean,
    b: [String, Boolean]
  }
}
</script>

<template>
  <pre>
a: {{a}}
b: {{b}}
  </pre>
</template>
```

```vue
<template>
  <MyComponent a b/>
  <MyComponent :a="true" :b="true"/>
</template>
```

Output:
![image](https://user-images.githubusercontent.com/16508807/151268439-f92ef195-2973-4aec-a470-63d52f4443b6.png)


So far, I haven't found a way to distinguish them well. If you have a good idea, please let me know.

https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBNeUNvbXBvbmVudCBmcm9tICcuL015Q29tcG9uZW50LnZ1ZSdcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxNeUNvbXBvbmVudCBhIGIvPlxuICA8TXlDb21wb25lbnQgOmE9XCJ0cnVlXCIgOmI9XCJ0cnVlXCIvPlxuPC90ZW1wbGF0ZT4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL3NmYy52dWVqcy5vcmcvdnVlLnJ1bnRpbWUuZXNtLWJyb3dzZXIuanNcIlxuICB9XG59IiwiTXlDb21wb25lbnQudnVlIjoiPHNjcmlwdD5cbmV4cG9ydCBkZWZhdWx0IHtcbiAgcHJvcHM6IHtcbiAgICBhOiBCb29sZWFuLFxuICAgIGI6IFtTdHJpbmcsIEJvb2xlYW5dXG4gIH1cbn1cbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxwcmU+XG5hOiB7e2F9fVxuYjoge3tifX1cbiAgPC9wcmU+XG48L3RlbXBsYXRlPiJ9

**Comment by ota-meshi** [comment]
If you change the order of types, both will give the same result.

```diff
<script>
export default {
  name: 'MyComponent',
  props: {
    a: Boolean,
-    b: [String, Boolean]
+    b: [Boolean, String]
  }
}
</script>
```

**Comment by FloEdelmann** [comment]
Duplicate of #775

**Comment by g-plane** [comment]
We can still report errors, but instead of providing auto fix, we can [provide suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions).

**Comment by ota-meshi** [comment]
> instead of providing auto fix, we can [provide suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions).

Sounds good to me!
